test_build_tag_push.py

```diff
@@ -0,0 +1,86 @@
+import unittest
+import build_tag_push
+import os
+import subprocess
+import argparse
+
+# args_app_yml_path = '/Users/alyssakelley/Desktop/test_mvp/ymltest.yml'
+# args_docker_hub_name = 'alyssakelley'
+
+# Question: as we talked about the parses before, I then tried to put them
+# in there own function and use them later on, but I still had to use the
+# special unittest.main line below. Should I take all of this away and
+# then just use hard coded test cases (like the path and docker_hub_name
+# as I have commented out above)?
+
+
+parser = argparse.ArgumentParser(description='Usage: python3'
+                                 'build_tag_push.py <docker_hub_name'
+                                 '<app_yml_path')
+parser.add_argument('docker_hub_name', type=str, help='DockerHub namespace')
+
+parser.add_argument('app_yml_path', type=str,
+                    help='Repository path to MVP Studio file')
+args = parser.parse_args()
+
+
+class TestBuildTagPush(unittest.TestCase):
+
+    def test_parse_app_file(self):  # Success.
+
+        data_from_app_yml = build_tag_push.parse_app_file(args.app_yml_path)
```

Comment:
The goal is that each test passes every time its run which you can't guarantee if the yml path is passed in. Instead you'd have something like the following directories *as part of this repo*:

```
test/data/
|--- build_fails_example/
|     |--- app.yml
|     |--- build.sh (this is a build script that returns a non-zero exit code simulating a failed build
|--- build_succeeds_example/
|     |--- app.yml
|     |--- build.sh (this build script returns a zero exit code)
|--- invalid_app_yml/
      |--- app.yml (this is invalid and won't parse)
```

Then, in *a single test* you'd do something like:

```
def test_throws_on_bad_yml(self):
    # note that while the build script will fail the yml is valid so parsing should be OK
    data = build_tag_push.parse_app_file('test/data/build_fails_example/app.yml')
    # Note that we know what's in the test data so we can make this assertion while if
    # the file comes from command line arguments you do *not* know that this will pass
    self.assertEqual(data['name'], 'build_fail_example')
    self.assertEqual(... etc...
```